src/s390x/zipl.rs

```diff
@@ -33,11 +34,116 @@ pub fn chreipl<P: AsRef<Path>>(dev: P) -> Result<()> {
     Ok(())
 }
 
+/// Secure boot (Secure IPL) includes support of SCSI and ECKD DASD boot devices
+enum Loaddev {
+    Eckd(String),
+    Scsi(String, String, String),
+}
+
+fn parse_lszdev_eckd(line: &str) -> Result<Loaddev> {
+    // ECKD ID looks like: 0.0.5223, we need only last part of it (5223)
+    lazy_static! {
+        static ref REGEX: Regex = Regex::new(r#"[[:digit:]].[[:digit:]].([[:xdigit:]]+)"#).unwrap();
+    }
+    if let Some(cap) = REGEX.captures_iter(line).next() {
+        return Ok(Loaddev::Eckd(cap[1].to_string()));
+    }
+    bail!("bad ECKD id: {}", line);
+}
```

Comment:
This is obviously totally fine, but just noting I think it's a one-liner to parse this without regexps:

```
line.split('.').nth(2).ok_or_else(|| anyhow!("Bad EKCD")).map(|s| Loaddev::Eckd(s.to_string()))
```
or so. (Though validation is looser)

Reply:
Yep, the reason to use regex here was to validate format of ID, which probably not that necessary.